frame/support/src/structs.rs

```diff
@@ -20,62 +20,13 @@
 use codec::{Decode, Encode, MaxEncodedLen};
 use scale_info::TypeInfo;
 // --- paritytech ---
-use frame_support::{
-	traits::{ConstU32, LockIdentifier},
-	WeakBoundedVec,
-};
-use pallet_balances::Reasons;
+use frame_support::{traits::ConstU32, WeakBoundedVec};
 use sp_runtime::{
 	traits::{AtLeast32BitUnsigned, Zero},
 	RuntimeDebug,
 };
 use sp_std::prelude::*;
 
-/// Frozen balance information for an account.
-pub struct FrozenBalance<Balance> {
-	/// The amount that `free` may not drop below when withdrawing specifically for transaction
-	/// fee payment.
-	pub fee: Balance,
-	/// The amount that `free` may not drop below when withdrawing for *anything except transaction
-	/// fee payment*.
-	pub misc: Balance,
-}
-impl<Balance> FrozenBalance<Balance>
-where
-	Balance: Copy + Ord + Zero,
-{
-	pub fn zero() -> Self {
-		Self { fee: Zero::zero(), misc: Zero::zero() }
-	}
-
-	/// The amount that this account's free balance may not be reduced beyond for the given
-	/// `reasons`.
-	pub fn frozen_for(self, reasons: Reasons) -> Balance {
-		match reasons {
-			Reasons::All => self.misc.max(self.fee),
-			Reasons::Misc => self.misc,
-			Reasons::Fee => self.fee,
-		}
-	}
-}
-
-/// A single lock on a balance. There can be many of these on an account and they "overlap", so the
-/// same balance is frozen by multiple locks.
-#[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug, MaxEncodedLen, TypeInfo)]
-pub struct OldBalanceLock<Balance, Moment> {
-	/// An identifier for this lock. Only one lock may be in existence for each identifier.
-	pub id: LockIdentifier,
-	pub lock_for: LockFor<Balance, Moment>,
-	/// If true, then the lock remains in effect even for payment of transaction fees.
-	pub reasons: Reasons,
-}
-
-#[derive(Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug, MaxEncodedLen, TypeInfo)]
-pub enum LockFor<Balance, Moment> {
-	Common { amount: Balance },
-	Staking(StakingLock<Balance, Moment>),
-}
-
 #[derive(Clone, Default, PartialEq, Eq, Encode, Decode, RuntimeDebug, MaxEncodedLen, TypeInfo)]
 pub struct StakingLock<Balance, Moment> {
```

Comment:
Why not move this to the staking pallet scope?

Reply:
Still in used.